cupy/functional/piecewise.py

```diff
@@ -0,0 +1,58 @@
+import cupy
+
+from cupy import core
+
+_piecewise_krnl = core.ElementwiseKernel(
+    'U condlist, S funclist',
+    'T y',
+    ' if(condlist) y = funclist',
+    'piecewise_kernel'
+)
+
+
+def piecewise(x, condlist, funclist):
+    """Evaluate a piecewise-defined function.
+
+        Args:
+            x (cupy.ndarray): input domain
+            condlist (list of cupy.ndarray or bool scalars):
+                Each boolean array/ scalar corresponds to a function
+                in funclist. Length of funclist is equal to that
+                of condlist. If one extra function is given, it is used
+                as otherwise condition
```

Comment:
The sentence `Length of funclist is equal to that of condlist. If one extra function is given, it is used as otherwise condition` is for funclist?

Reply:
If one extra function is given, it is used as the default value when the otherwise condition is met